tests/PHPStan/Analyser/data/bug-7344.php

```diff
@@ -0,0 +1,87 @@
+<?php
+
+namespace Bug7344;
+
+use Exception;
+use function PHPStan\Testing\assertType;
+
+interface PhpdocTypeInterface
+{
+    /**
+     * Interface for phpdoc type for Phpstan only
+     *
+     * @return never
+     */
+    public function neverImplement(): void;
+}
+
+interface IsEntity extends PhpdocTypeInterface
+{
+}
+
+class Model
+{
+
+	/**
+	 * @return static
+	 * @phpstan-return static&IsEntity
+	 */
+	public function createEntity(): self
+	{
+		throw new Exception();
+	}
+
+	/**
+	 * @return static
+	 */
+	public function getModel(bool $allowOnModel = false): self
+	{
+		throw new Exception();
+	}
+
+}
+
+class Female extends Person
+{
+}
+
+class Male extends Person
+{
+}
+
+// DEBUG
+abstract class FemaleWithImplementedInterfaceForDebug extends Female implements IsEntity
+{
+	public function neverImplement(): void
+	{
+		throw new Exception();
+	}
+}
+
+class Foo
+{
+
+	public function doFoo()
+	{
+//		// simple type
+//		$model = new Model();
+//		assertType('Bug7344\Model', $model);
+//
+//		$entity = $model->createEntity();
+//		assertType('Bug7344\IsEntity&Bug7344\Model', $entity);
+//		assertType('Bug7344\Model', $entity->getModel());
+//		assertType('Bug7344\Model', random_int(0, 1) === 0 ? $model : $entity);
+//
+//		// union type
+//		$modelClass = random_int(0, 1) === 0 ? Female::class : Male::class;
+//		$model = new $modelClass();
+//		assertType('Bug7344\Female|Bug7344\Male', $model);
+
+		/** @var \Bug7344\IsEntity&\Bug7344\Female */
+		$entity = $model->createEntity();
+		assertType('Bug7344\Female&Bug7344\IsEntity', $entity);
+		assertType('Bug7344\Female|Bug7344\Male', $entity->getModel());
```

Comment:
@ondrejmirtes here I get `*ERROR*` but the `PHPStan\Tests\Bug7344DynamicReturnTypeExtension::getTypeFromMethodCall()` seems not even called - why? And is it possible to dump the cause/exception of `*ERROR*` in the test easily?

Reply:
yes, thanks, I was confused why I did different results than on the original issue... will update shortly